python-checks/src/main/java/org/sonar/python/checks/regex/UnusedGroupNamesCheck.java

```diff
@@ -60,23 +59,25 @@ public class NotExistedGroupUsagesCheck extends AbstractRegexCheck {
   private static final Set<String> COMPILE_FUNCTION_NAMES = Set.of("re.compile");
 
 
-  public static final String GROUP_NAME_DOESNT_EXISTS_MESSAGE_FORMAT = "There is no group named '%s' in the regular expression.";
-  public static final String GROUP_NUMBER_DOESNT_EXISTS_MESSAGE_FORMAT = "There is no group with number '%d' in the regular expression.";
+  private static final String GROUP_NAME_DOESNT_EXISTS_MESSAGE_FORMAT = "There is no group named '%s' in the regular expression.";
+  private static final String USE_NAME_INSTEAD_OF_NUMBER_MESSAGE_FORMAT = "Directly use '%s' instead of its group number.";
+  private static final String GROUP_NAME_SECONDARY_MESSAGE_FORMAT = "Named group '%s'";
+  private static final String GROUP_NUMBER_SECONDARY_MESSAGE_FORMAT = "Group %d";
+  private static final String NO_GROUP_NAMES_SECONDARY_MESSAGE = "No named groups defined in this regular expression.";
 
   @Override
   public void checkRegex(RegexParseResult regexParseResult, CallExpression regexFunctionCall) {
     var groupsCollector = new KnownGroupsCollector();
     groupsCollector.visit(regexParseResult);
 
     getCallExpressionResultUsages(regexFunctionCall, COMPILE_FUNCTION_NAMES)
```

Comment:
(Not asking for change): I kind of have the feeling that it would have been more clear if we checked the called method upfront and branched depending on that (in a simple imperative `if/else`). Then, unconditionally retrieve usages in `getCallExpressionResultUsages` and extracted the group accesses in both branches.
Since we're late, I don't think it's worth refactoring this, as it is clear enough anyway. Just sharing a personal preference, please don't change this.